### PR TITLE
CI/CD workflows + Docker setup + Code formatting

### PR DESCRIPTION
**Pull request by HarshCasper**
## Description

This PR:

- [format code using black](https://github.com/interviewstreet/ghs/commit/bc0f5ab747b9b8de51623e15069ae98cb943285a)
- [add flake8 configuration](https://github.com/interviewstreet/ghs/commit/cf54a8aeed3caa9c90e779c29af6d6aab6b700c2)
- [add a gha workflow to build & lint](https://github.com/interviewstreet/ghs/commit/d4753d0d5e762b5f4ff78095334a3b8865741798)
- [add dockerfile and instructions to build and use](https://github.com/interviewstreet/ghs/commit/f1387140339e7d73608afee659c3a68c4b3ffaf7)

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Upgrade pip version
+        run: |
+          python3 -m pip install --upgrade pip
+
+      - name: Install the project
+        run: |
+          pip install .
```
Comment (makkoncept) [inline .github/workflows/ci.yml:38]: Just to be consistent with other commands, can we change this to `pip3`?

**Comment by makkoncept** [comment]
Thank you @HarshCasper for the contribution!

1. The project was already using _autopep8_ for formatting (it's also mentioned in [CONTRIBUTING.md](https://github.com/interviewstreet/ghs/blob/master/CONTRIBUTING.md#code-formatting)). But I guess we can adopt black because it's has sensible defaults which takes out the pain of constant configuration and also black is widely adopted by the python community. 
2. Cool. Can you update the [CONTRIBUTING.md#coding-style](https://github.com/interviewstreet/ghs/blob/master/CONTRIBUTING.md#coding-style) to mention that we use _black_ for code formatting and _flake8_  as the linter.
3. Cool. Added a minor comment above.
4. I don't think that dockerizing the package has any advantages. If the idea is to isolate the installation and don't affect the host machine, then [pipx](https://pypa.github.io/pipx/) is more straightforward than this (also, the good old virtual environment can also be used). Though I don't have any strong opinions on this. Would love to hear your thoughts on this point.

**Comment by Rohit484** [comment]
https://github.com/Rohit484

File: .github/workflows/ci.yml (hunk 1 above)
Comment (HarshCasper) [reply .github/workflows/ci.yml]: Done!

**Comment by HarshCasper** [comment]
Hi @makkoncept 

I have made all the recommended changes. Coming to Dockerizing the package, I think it's a good addition to kick-start the CLI faster and reliably without having the user deal with virtual environments and making it ubiquitous to use. We can also offer an extended Docker image over GHCR/DockerHub which the users can pull and use without having to clone the repository or install the Python package.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -36,6 +36,8 @@ Contribution can be anything like (but not limited to) improving documentation,
 
 ### Coding Style
 
+This project uses the [Black](https://black.readthedocs.io/en/stable/) code formatter for formatting the code. [Flake8](https://flake8.pycqa.org/en/latest/) is used for linting the code and validating the style and structure.
+
 #### Code Formatting
 
 - The project uses _autopep8_ as the code-formatter.
```
Comment (makkoncept) [inline CONTRIBUTING.md:43]: Nitpick: Can you remove Line 41-43 as they are not redundant.

**Review by makkoncept — CHANGES_REQUESTED** [review]

**Pull request merged**